akka-docs/src/main/paradox/typed/cluster-singleton.md

```diff
@@ -74,6 +74,19 @@ Be aware that this means there will be times when the singleton won't be running
 See @ref[Fault Tolerance](./fault-tolerance.md) for a full list of supervision options.
 
 
+## Custom stop message
+
+Singleton's are migrated to a new node if the node they are running on is shut down. If asynchronous shutdown logic needs to be executed
+then a custom stop message can be used. Upon receiving the stop message it is then the responsibility of the actor to shutdown. 
```

Comment:
Is this not the same `terminationMessage: Any` behavior in the untyped `akka.cluster.singleton.ClusterSingletonManager`? 

If it is, why duplicate and differentiate docs?
* https://github.com/akka/akka/blob/master/akka-docs/src/main/paradox/cluster-singleton.md#an-example
* https://github.com/akka/akka/blob/master/akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala#L420-L426)

I like this wording, perhaps unless typed does it differently, we should have one cohesive message around the functionality. Can you merge this with the existing code/doc verbiage so we are always only saying the same thing, if this is the case?

Reply:
The docs for typed are separate from untyped in a very unplanned and unstructured way. The end goal is to remove/archive the untyped docs so I don't want to link from typed to untyped. I hope this will be discussed at our next meeting. 

I'll align the terminology though as much asp possible but I'd rather stick with `stopMessage` than `terminationMessage` to be consistent with typed sharding rather than untyped singleton. Typed can't have poison pill as its stop message as it isn't part of the protocol and `PostStop` signal is just for typed.

Reply:
Thanks @chbatey, this is an issue I am seeing around the typed codebase, I'll add it to the agenda for the, next meeting as part of the overarching typed topic.

Reply:
We had it in the last meeting but we never got to it sadly :(